callisto/delivery/api.py

```diff
@@ -0,0 +1,34 @@
+from django.conf import settings
+from django.utils.module_loading import import_string
+
+
+class DeliveryNotificationApi(object):
+    '''
+        Used to route calls to from inside of callisto/delivery/* to
+        notification api(s) inside of other apps
+
+        this object is called with
+            from callisto.delivery.api import DeliveryNotificationApi
+            DeliveryNotificationApi().example_call
+
+        which maps to (on default settings)
+            from callisto.notification.api import NotificationApi
+            NotificationApi.example_call
+
+        use CALLISTO_NOTIFICATION_API to override the class that
+        calls to DeliveryNotificationApi map to
+    '''
+
+    def __init__(self):
+        self.set_api_class()
+
+    def __getattr__(self, attr):
+        return getattr(self.api, attr)
+
+    def set_api_class(self):
+        override_class_path = getattr(
+            settings,
+            'CALLISTO_NOTIFICATION_API',
+            'callisto.notification.api.NotificationApi',
```

Comment:
I'd prefer to separate the two concerns (defining the interface, and routing calls) into two separate classes. Then you can make the interface class an [abstract base class](https://zaiste.net/2013/01/abstract_classes_in_python/), which means that an incomplete re-implementation will fail at instantiation rather than call-time, which I think might be nice given that these methods are called fairly rarely (plus I think you might think it's fun...)